dali/c_api_2/pipeline.cc

```diff
@@ -0,0 +1,302 @@
+    FeedInputImpl(
+      input_name,
+      *input_data->Unwrap<GPUBackend>(),
+      data_id,
+      options,
+      stream);
+  } else {
+    assert(!"Impossible device type encountered.");
+  }
+}
+
+int PipelineWrapper::GetOutputCount() const {
+  return pipeline_->output_descs().size();
+}
+
+daliPipelineOutputDesc_t PipelineWrapper::GetOutputDesc(int idx) const {
+  auto &outputs = pipeline_->output_descs();
+  int nout = outputs.size();
+  if (idx < 0 || idx >= nout)
+    throw std::out_of_range(make_string(
+        "The output index ", idx, " is out of range. "
+        "Valid range is [0..", nout-1, "]."));
+  auto &out = outputs[idx];
+  daliPipelineOutputDesc_t desc{};
+  desc.device         = static_cast<daliStorageDevice_t>(out.device);
+  desc.dtype          = out.dtype;
+  desc.dtype_present  = out.dtype != DALI_NO_TYPE;
+  desc.name           = out.name.c_str();
+  desc.ndim           = out.ndim;
+  desc.ndim_present   = out.ndim >= 0;
+  return desc;
+}
+
+template <typename Backend>
+void PipelineWrapper::FeedInputImpl(
+      std::string_view input_name,
+      const TensorList<Backend> &tl,
+      std::optional<std::string_view> data_id,
+      daliFeedInputFlags_t options,
+      std::optional<cudaStream_t> stream) {
+  InputOperatorCopyMode copy_mode = InputOperatorCopyMode::DEFAULT;
+
+  if (options & DALI_FEED_INPUT_FORCE_COPY) {
+    if (options & DALI_FEED_INPUT_NO_COPY)
+      throw std::invalid_argument("DALI_FEED_INPUT_FORCE_COPY and DALI_FEED_INPUT_NO_COPY"
+                                  " must notbe used together.");
+    copy_mode = InputOperatorCopyMode::FORCE_COPY;
+  } else if (options & DALI_FEED_INPUT_NO_COPY) {
+    copy_mode = InputOperatorCopyMode::FORCE_NO_COPY;
+  }
+
+  pipeline_->SetExternalInput(
+    std::string(input_name),  // TODO(michalz): switch setting input to string_view
+    tl,
+    stream.has_value() ? AccessOrder(*stream) : tl.order(),
+    options & DALI_FEED_INPUT_SYNC,
+    options & DALI_FEED_INPUT_USE_COPY_KERNEL,
+    copy_mode,
+    data_id ? std::optional<std::string>(std::in_place, *data_id) : std::nullopt);
+}
+
+}  // namespace dali::c_api
+
+using namespace dali::c_api;  // NOLINT
+
+daliResult_t daliPipelineCreate(
+      daliPipeline_h *out_pipe_handle,
+      const daliPipelineParams_t *params) {
+  DALI_PROLOG();
+  CHECK_OUTPUT(out_pipe_handle);
+  NOT_NULL(params);
+
+  *out_pipe_handle = new PipelineWrapper(*params);
+  DALI_EPILOG();
+}
+
+daliResult_t daliPipelineDestroy(daliPipeline_h pipeline) {
+  DALI_PROLOG();
+  delete ToPointer(pipeline);
+  DALI_EPILOG();
+}
+
+daliResult_t daliPipelineDeserialize(
+      daliPipeline_h *out_pipe_handle,
+      const void *serialized_pipeline,
+      size_t serialized_pipeline_size,
+      const daliPipelineParams_t *param_overrides) {
+  DALI_PROLOG();
+  CHECK_OUTPUT(out_pipe_handle);
+  NOT_NULL(serialized_pipeline);
+  NOT_NULL(param_overrides);
+
+  *out_pipe_handle = new PipelineWrapper
+      (serialized_pipeline,
```

Comment:
```suggestion
  *out_pipe_handle = new PipelineWrapper(
        serialized_pipeline,
```
I know this will be controversial opinion, by IMO we should be running fully new code (especially one that is external-user facing in the headers) through our clang-format config, for consistency of formatting and increased readability thanks to that.

Reply:
I wonder where this formatting came from and how it even got past the linter...